src/plugins/es_ui_shared/README.md

```diff
@@ -0,0 +1,33 @@
+## ES UI shared modules
+
+This plugin contains reusable code in the form of self-contained modules
+(or libraries). Each of these modules exports a set of functionality
+relevant to the domain of the module.
+
+## Files and folders overview
+
+- `./public` | `./server`. Folders for grouping server or public code according to the Kibana plugin pattern.
+- `./__packages_do_not_import__` is where actual functionality is kept. This enables modules more control over what functionality is directly exported and prevents parts of modules to be depended on externally in unintended ways.
+- `./public/index.ts` | `./server/index.ts` These files export modules (simple JavaScript objects). For example, `Monaco` is the name of a module. In this way, modules namespace all of their exports and do not have to be concerned about name collisions from other modules.
+
+## Conventions for adding code
+
+When adding new functionality, look at the folders in `./__packages_do_not_import__` and consider whether your functionality falls into any of those
+modules.
+
+If it does not, you should create a module and expose it to public or server code (or both) following the conventions described above.
+
+### Example
+
+If I wanted to add functionality for calculating a Fibonacci sequence browser-side one would do the following:
```

Comment:
minor grammar note: "If **one** wanted to ... **one** would do the following ..."